spotlight-client/src/DataStore/TenantStore.ts

```diff
@@ -41,6 +42,14 @@ export default class TenantStore {
     this.rootStore = rootStore;
 
     this.tenants = new Map();
+
+    // tenant mapped from domain should be locked
+    const tenantFromDomain = getTenantFromDomain();
+    if (tenantFromDomain) {
+      this.currentTenantId = tenantFromDomain;
+      // returning null renders an observable property immutable
+      intercept(this, "currentTenantId", () => null);
```

Comment:
Does this `intercept` basically revoke the observable property, so that `this.currentTenantId` is only an observable property until it's set the first time, and then it's immutable?

Reply:
sort of ... it's still observable, what [`intercept`](https://mobx.js.org/intercept-and-observe.html#intercept) does here is observe all the incoming changes and discard them before they can be applied. So the original value remains because we set it before registering the interceptor